ludwig/hyperopt/run.py

```diff
@@ -373,3 +404,36 @@ def update_hyperopt_params_with_defaults(hyperopt_params):
         hyperopt_params[EXECUTOR],
         executor_defaults,
     )
+
+
+def get_shared_params_dict(config_feature_type: str, features: Dict[str, Any]) -> Dict[str, Dict[str, Set]]:
+    """Generates a mapping of feature type to the corresponding set of features without an encoder or one using the
+    default encoder for that feature type.They may be considered for potential shared parameter search spaces
+    depending on the parameter space defined later within the hyperopt config.
+
+    This applies to both config_feature_types - input_features and output_features. The shared parameters for both
```

Comment:
Can we add somewhere:

```
Note that shared default parameter search spaces are not applied to features with non-default encoders or non-default decoders, since shared default parameter values should only apply to default modules.
```